src/main/java/xyz/nucleoid/bedwars/game/active/shop/BwTeamShop.java

```diff
@@ -9,84 +12,76 @@
+                Text hasteName = new TranslatableText("team_upgrade.bedwars.haste");
                 shop.add(ShopEntry.ofIcon(Items.GOLDEN_PICKAXE)
-                        .withName(new LiteralText("Activate Haste"))
-                        .addLore(new LiteralText("All team members will get"))
-                        .addLore(new LiteralText("a permanent haste effect"))
-                        .withCost(hasteCost)
+                        .withName(hasteName)
+                        .addLore(new TranslatableText("team_upgrade.bedwars.haste.description"))
+                        .withCost(!teamState.hasteEnabled ? Cost.ofDiamonds(3) : Cost.no())
                         .onBuy(p -> {
                             teamState.hasteEnabled = true;
-                            game.broadcast.broadcastTeamUpgrade(participant, new LiteralText("activated haste!"));
+                            game.broadcast.broadcastToTeam(participant.team, new TranslatableText("team_upgrade.bedwars.haste.buy", p.getDisplayName().shallowCopy()).formatted(Formatting.BOLD, Formatting.AQUA));
                         })
                 );
 
                 int sharpness = teamState.swordSharpness;
                 int nextSharpness = Math.min(sharpness + 1, BwActive.TeamState.MAX_SHARPNESS);
 
-                Cost sharpnessCost = sharpness != nextSharpness ? Cost.ofDiamonds(stagedUpgrade(4, sharpness)) : Cost.no();
+                Text sharpnessName = new TranslatableText("team_upgrade.bedwars.sharpness", new TranslatableText("enchantment.level." + nextSharpness));
                 shop.add(ShopEntry.ofIcon(Items.DIAMOND_SWORD)
-                        .withName(new LiteralText("Sword Sharpness " + nextSharpness))
-                        .addLore(new LiteralText("All team members will get"))
-                        .addLore(new LiteralText("sharpness applied to their swords"))
-                        .withCost(sharpnessCost)
+                        .withName(sharpnessName)
+                        .addLore(new TranslatableText("team_upgrade.bedwars.sharpness.description", new TranslatableText("enchantment.level." + nextSharpness)))
+                        .withCost(sharpness != nextSharpness ? Cost.ofDiamonds(stagedUpgrade(4, sharpness)) : Cost.no())
                         .onBuy(p -> {
                             teamState.swordSharpness = Math.max(nextSharpness, teamState.swordSharpness);
                             game.teamLogic.applyEnchantments(participant.team);
-                            game.broadcast.broadcastTeamUpgrade(participant, new LiteralText("added Sword Sharpness " + teamState.swordSharpness));
+                            game.broadcast.broadcastToTeam(participant.team, new TranslatableText("team_upgrade.bedwars.sharpness.buy", p.getDisplayName().shallowCopy(), new TranslatableText("enchantment.level." + teamState.swordSharpness)).formatted(Formatting.BOLD, Formatting.AQUA));
```

Comment:
General commend for all of the entries here: we should try and keep away from hardcoding further like has been done here to allow for future logic, such as variable shop prices depending on player count